SimPEG/utils/io_utils/io_utils_electromagnetics.py

```diff
@@ -1,35 +1,208 @@
+    file_name: Path to xyz file
+    data_type: ["volt", "apparent_resistivity", "apparent_chargeability"]
+    a_headers: A list or tuple of strings representing the headers of the A-electrode location
+    b_headers: A list or tuple of strings representing the headers of the B-electrode location
+    m_headers: A list or tuple of strings representing the headers of the M-electrode location
+    n_headers: A list or tuple of strings representing the headers of the N-electrode location
+    data_header: String representing the header for the data column
+    uncertainties_header: String representing the header for the data column
+    dict_headers: list or tuple of strings for additional data columns being loaded into a dictionary
+    is_surface_data: True if surface formatted and electrode elevations are not supplied
+    """
+    
+    if data_type not in ["volt", "apparent_resistivity", "apparent_chargeability"]:
+        raise Exception(
+            "data_type must be one of 'volt', 'apparent_resistivity', 'apparent_chargeability'"
+        )
+    
+    # Prevent circular import
+    from ...electromagnetics.static import resistivity as dc
+    from ...data import Data
+    
+    # Load file headers
+    FID = open(file_name, 'r')
+    file_headers = FID.readline()
+    FID.close()
+    file_headers = file_headers.split()
+
+    # Find indices of columns being loaded
+    out_headers = a_headers + b_headers + m_headers + n_headers
+    
+    has_data = False
+    has_uncert = False
+    has_dict = False
+    non_location_columns = 0
+    if data_header is not None:
+        out_headers.append(data_header)
+        has_data = True
+        non_location_columns += 1
+    if uncertainties_header is not None:
+        out_headers.append(uncertainties_header)
+        has_uncert = True
+        non_location_columns += 1
+    if dict_headers is not None:
+        out_headers += dict_headers
+        has_dict = True
+        non_location_columns += len(dict_headers)
+    
+    col_indices = []
+    for h in out_headers:
+        col_indices.append(file_headers.index(h))
+    
+    # Load specified columns of data
+    data_array = np.loadtxt(
+        file_name, comments='!', skiprows=1, usecols=col_indices
+    )
+    n_rows = np.shape(data_array)[0]
+    num_location_columns = np.shape(data_array)[1] - non_location_columns
+    
+    # Extract columns for electrode locations
+    if num_location_columns == 12:
+        a_cols = [0, 1, 2]
+        b_cols = [3, 4, 5]
+        m_cols = [6, 7, 8]
+        n_cols = [9, 10, 11]
+    # 2D survey data or surface 3D data
+    elif num_location_columns == 8:
+        a_cols = [0, 1]
+        b_cols = [2, 3]
+        m_cols = [4, 5]
+        n_cols = [6, 7]
+    # 2D surface data
+    else:
+        a_cols = [1]
+        b_cols = [2]
+        m_cols = [3]
+        n_cols = [4]
+    
+    # Extract electrode locations
+    if is_surface_data:
+        dummy_elevation = 9999  # Taller than mount Everest
```

Comment:
just set it as like, `np.inf` or `np.nan`

Reply:
I thought about it, but I like this better. Unless we want to make it impossible for the user to apply any operations until proper elevations are applied to the data.